### PR TITLE
Fix: Add `random_state=42` to `train_test_split()` calls for reproducibility

### LINKED ISSUE #1407: Fix: Add random_state to train_test_split calls for reproducible ML notebook results
## Description

Several machine learning notebooks in the repository use train_test_split() without specifying a random_state parameter.

Because of this, the train-test split changes every time the notebook is executed, which causes:

- inconsistent accuracy scores,
- non-reproducible outputs,
- unstable model evaluation,
- and confusion for beginners following the notebooks.

For an educational ML repository, reproducibility is extremely important so that learners can consistently obtain the same results while studying or experimenting with the notebooks.

### Current Behavior

- Example of the current implementation:

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)

Since no random_state is provided, the dataset split changes on every execution.

### Proposed Solution

Add a fixed random_state value (random_state=42) to all affected train_test_split() calls.

- Updated implementation:

X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42
)
### Notebooks Targeted

The fix will be applied to the affected notebooks where reproducibility is currently missing, including notebooks related to:

- Heart Disease Prediction
- Diabetes ML notebooks
- Flood Prediction
- Language Detection
- Startup Profit Prediction
- SVM examples
- GridSearchCV examples

and other confirmed notebooks with missing random_state.

### Why This Fix Is Important

Adding random_state improves:

- reproducibility of experiments
- consistency of notebook outputs
- debugging reliability
- educational clarity for beginners
- overall ML workflow stability

This is a standard best practice in machine learning workflows and educational repositories.

## Additional Notes

Only reproducibility-related changes will be made.
No dataset, model architecture, or training logic will be modified.
Notebook outputs will be cleared before submission to keep the PR clean.

I would like to work on this issue under GSSoC 2026 open source contribution.

### PR DESCRIPTION
**Pull request by Yug-Katyal**
### Overview

This PR improves the reproducibility of several machine learning notebooks by adding a fixed `random_state=42` to `train_test_split()` calls where it was previously missing.

Without a fixed random state, the train-test split changes on every execution, resulting in different model performance metrics and making it difficult for users to reproduce the notebook results.

## Current Behavior

Many notebooks currently use:

```python
X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2
)
```

Since `random_state` is not specified, each execution generates a different train-test split, leading to inconsistent evaluation results.

## Updated Changes

Updated the affected `train_test_split()` calls to include a fixed random seed:

```python
X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42
)
```

Only calls **without an existing `random_state`** were modified. Existing `random_state` values (e.g., `0`, `1`, etc.) were left unchanged.

## Benefits

* Improves reproducibility of experiments
* Ensures consistent model evaluation across runs
* Makes debugging and result verification easier
* Provides a better learning experience for beginners
* Follows standard machine learning best practices

Fixes #1407

### CODE REVIEW

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
remove the workflows folder @Yug-Katyal 
it's under development


after making changes to the notebooks u have to re run all the cells @Yug-Katyal
